### PR TITLE
Implement ListAttendance command and parser with tests

### LINKED ISSUE #140: Implement ListAttendance command and parser with tests
Add a 'listattendance' command to let tutors view a student's attendance records from the command result display.

Tasks:
- Create 'ListAttendanceCommand' to retrieve a student by displayed index and format attendance records as readable text
- Support optional filtering by subject with 's/SUBJECT'
- Create 'ListAttendanceCommandParser' to parse 'listattendance INDEX [s/SUBJECT]'
- Register the command in 'AddressBookParser'
- Add tests for valid and invalid command/parser behavior

### PR DESCRIPTION
**Pull request by daynerss1**
Implements the new 'listattendance' command for viewing a student's attendancerecords in the command result display, with optional filtering by subject.

Tasks:
- Added 'ListAttendanceCommand' to retrieve a student by displayed index and format attendance records as readable text
- Added optional subject filtering with 's/SUBJECT'
- Added 'ListAttendanceCommandParser' for 'listattendance INDEX [s/SUBJECT]'
- Registered the command in 'AddressBookParser'
- Added command and parser tests for success and edge cases

Fixes #140

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ListAttendanceCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListAttendanceCommand.MESSAGE_USAGE), pe);
+        }
+
+        String subject = null;
+        if (argMultimap.getValue(PREFIX_SUBJECT).isPresent()) {
+            subject = ParserUtil.parseSubject(argMultimap.getValue(PREFIX_SUBJECT).get()).subjectName;
```
Comment (stinkray77) [inline src/main/java/seedu/address/logic/parser/ListAttendanceCommandParser.java]: You are calling .subjectName on the result of ParserUtil.parseSubject(...). Ensure that parseSubject is throwing the appropriate ParseException if the subject is invalid, as the parser relies on that to provide feedback to the user via the catch block in AddressBookParser.
